File: ClientApp/src/models/entities/identity/connection.ts
import {Entity} from "../entity";
import {User} from "./user.entity";

export class Connection extends Entity<number> {
  browser: string;
  remoteAddress: string;
  os: string;
  isPersistent: boolean;
  beginDate: Date;
  endDate: Date;

  user: User;
  userId: string;

  get isClosed() {
    return this.endDate != null;
  }

  static createFromAny(value: any) : Connection {
    if(!value) {
      return null;
    }

    const connection = new Connection();

    connection.id = value.id;
    connection.registrationDate = value.registrationDate;

    connection.browser = value.navigator;
    connection.remoteAddress = value.remoteAddress;
    connection.os = value.os;
    connection.isPersistent = value.isPersistent;
    connection.beginDate = value.beginDate;
    connection.endDate = value.endDate;

    connection.user = User.createFromAny(value.user);
    connection.userId = value.userId;

    return connection;
  }
}

File: ServerApp/Entities/Identity/Connection.cs
using System;
using Everest.AspNetStartup.Binding;
using Everest.AspNetStartup.Models;
using Microsoft.AspNetCore.Mvc;

namespace Exam.Entities.Identity
{
    [ModelBinder(BinderType = typeof(ItemValueModelBinder))]
    public class Connection : Entity<long>
    {
        public string Browser { get; set; }

        public string RemoteAddress { get; set; }

        public string OperatingSystem { get; set; }
        
        public bool IsPersistent { get; set; }

        public string Type { get; set; }

        public bool Online { get; set; }
        
        public string Position { get; set; }
        public DateTime? BeginDate { get; set; }
        public DateTime? EndDate { get; set; }

        public bool IsClosed { get => EndDate != null; }

        
        public virtual User User { get; set; }
        public string UserId { get; set; }
    }
}